### ISSUE TITLE
Falcon7b prefill remaining fixes and cleanup after enabling optimized version

### ISSUE THREAD

**Issue by s-jovic** [tenstorrent/tt-metal#8349]
- [x] Segfault for 1k and 2k prefill on multi-device (single run) and single device (when run in a loop): https://github.com/tenstorrent/tt-metal/issues/8644 (worked around)
- [x] Async mode (not all ops used in optimized prefill support async mode)
- [ ] Make model initialization agnostic to sequence length
- [ ] GS path - decide whether to use the optimized version for GS and if so make it work
- [ ] Clean up lm head - move configuration to `model_config` file
- [ ] Push 128seq len output PCC to 0.99: [[issue](https://github.com/tenstorrent/tt-metal/issues/8487)](https://github.com/tenstorrent/tt-metal/issues/8487)
- [ ] Reduce memory usage of e2e tests - the 2k test uses up to 80gb ram for some reason; try to run PyTorch and tt model sequentially

**Comment by pavlepopovic** [comment]
- [ ] Check that all MMs have optimal settings (sunblock_h/w,, in0_block_w) following di/dt fixes (attention, MLP, LM head matmuls)
- [ ] Check if any more sharding is possible throughout model

**Comment by s-jovic** [comment]
- [ ] remove persistent kernel cache usage
- [ ] put optimized attention on CI
- [ ] unify paths for 128/1024/2k and other sequence lengths
- [ ] perf breakdown

**Comment by pavlepopovic** [comment]
- [ ] Use bfp8/bfp4 for attention mask. Currently we have softmax per slice running at ~350k ns. Using bfp8 attention mask brings that down to ~186k, and bfp4 further optimises it to ~171k. This improves attention by ~2ms/layer @2k seq len.

**Comment by pavlepopovic** [comment]
- [ ] Use sharded layernorm instead of layernorm->bcast->bcast
